### ISSUE TITLE
mix.js file not deleted where other methods called before js()

### ISSUE THREAD

**Issue by tomcastleman** [JeffreyWay/laravel-mix#687]
- Laravel Mix Version: 0.10.0
- Node Version: v6.10.0
- NPM Version: 3.10.10
- OS: macOS Sierra 10.12.4

### Description:

When calling mix in this order (default in Spark):
```
mix.sass('resources/assets/sass/v2.scss', 'public/css')
  .js('resources/assets/js/app.js', 'public/js')
  .version();
```

There is a versioned `public/mix.[hash].js` left behind on the filesystem and in `mix-manifest.json`:
```
{
  "/mix.js": "/mix.53cf2ec0a72208b30e4e.js",
  "/css/v2.css": "/css/v2.10d79924a4d6add686b8.css",
  "/js/foo.js": "/js/foo.eb4bc905a2df30cc56ee.js"
}
```

... and output of `npm run dev`:
```
 DONE  Compiled successfully in 6974ms                                                                                                                                                                                                             8:52:37 PM
                                                             Asset       Size  Chunks                    Chunk Names
...
                                       mix.53cf2ec0a72208b30e4e.js    3.03 kB       0  [emitted]         mix
                                   /js/foo.eb4bc905a2df30cc56ee.js    2.82 kB       1  [emitted]         /js/foo
                                  /css/v2.10d79924a4d6add686b8.css     756 kB       0  [emitted]  [big]  mix
                                                 mix-manifest.json  151 bytes          [emitted]
```

When placing the `.js()` call first, this doesn't happen:
```
mix.js('resources/assets/js/foo.js', 'public/js').
  .sass('resources/assets/sass/v2.scss', 'public/css')
  .version();
```

```
DONE  Compiled successfully in 6869ms                                                                                                                                                                                                             8:57:12 PM

                                                             Asset       Size  Chunks                    Chunk Names
...
                                   /js/foo.4926749fa53a0402c48b.js    3.01 kB       0  [emitted]         /js/foo
                                  /css/v2.02107478d4dd2b2b628b.css     756 kB       0  [emitted]  [big]  /js/foo
                                                 mix-manifest.json  106 bytes          [emitted]
```

This also occurs without the `version()` call (a `mix.js` file is left on disk and in manifest with no hash).

### Steps To Reproduce:

As above

### Question:

I gather from [#572](https://github.com/JeffreyWay/laravel-mix/issues/572) that this is potentially expected behaviour, but I think the file should be being deleted.

Should the order of the method calls matter, or is there a bug to be fixed here?

Thanks

**Comment by aaronhuisinga** [comment]
I was wondering about this, as well. It seems like a bug to me.

**Comment by JonasDoebertin** [comment]
Happening here, too.

**Comment by ekandreas** [comment]
Same here! A lot of versions left in the dist folder :-)

**Comment by steveheinsch** [comment]
I just upgraded and this is happening on v0.12 as well (also using Spark if that matters)

**Comment by JeffreyWay** [comment]
This is fixed in Mix 1.0.
